custom/ilsgateway/tanzania/handlers/zipline.py

```diff
@@ -0,0 +1,66 @@
+from corehq.apps.products.models import SQLProduct
+from custom.ilsgateway.tanzania.handlers.keyword import KeywordHandler
+from custom.ilsgateway.tanzania.reminders import INVALID_PRODUCT_CODE
+from custom.zipline.api import ProductQuantity
+
+
+class ParseError(Exception):
+    pass
+
+
+class ZiplineGenericHandler(KeywordHandler):
+
+    help_message = None
+    error_message = None
+
+    def _check_quantities(self, quantities):
+        for quantity in quantities:
+            try:
+                int(quantity)
+            except ValueError:
+                raise ParseError(self.error_message)
+
+    def _check_product_codes(self, product_codes):
+        for product_code in product_codes:
+            try:
+                SQLProduct.objects.get(code=product_code, domain=self.domain)
+            except SQLProduct.DoesNotExist:
+                raise ParseError(INVALID_PRODUCT_CODE % {'product_code': product_code})
+
+    def parse_message(self, text):
+        text_split = text.split()
+        product_codes = text_split[::2]
+        quantities = text_split[1::2]
+
+        product_codes_len = len(product_codes)
+        if product_codes_len == 0 or product_codes_len != len(quantities):
+            raise ParseError(self.error_message)
+        self._check_quantities(quantities)
+        self._check_product_codes(product_codes)
+        return zip(product_codes, quantities)
+
+    def help(self):
+        self.respond(self.help_message)
+        return True
+
+    def send_success_message(self):
+        raise NotImplementedError()
+
+    def invoke_api_function(self, quantities_list):
+        raise NotImplementedError()
+
+    def handle(self):
```

Comment:
@kkrampa i just realized, we don't want this to be live for the production domain yet. can you add a toggle named ZIPLINE which checks here whether or not it's enabled for the domain and if it's not it just returns